stablehlo/dialect/TypeInference.cpp

```diff
@@ -369,6 +400,40 @@ LogicalResult verifyAddOp(std::optional<Location> location, Operation* op,
   return success();
 }
 
+LogicalResult verifyTransposeOp(std::optional<Location> location,
+                                Type operandType, ArrayRef<int64_t> permutation,
+                                Type resultType) {
+  // transpose_c1
+  if (allQuantized<quant::UniformQuantizedType>(operandType, resultType)) {
+    if (getElementTypeOrSelf(operandType) != getElementTypeOrSelf(resultType))
+      return emitOptionalError(location,
+                               "expect same quantization scale and zero_point "
+                               "for operand and result but got ",
+                               operandType, " vs ", resultType);
+  }
```

Comment:
Lots of repeating logic. I think this PR can benefit from a few functions like:

```
// Verifies uniform quantized and per-axis scale and zero point match
LogicalResult verifyQuantizedScaleAndZeroPointConstraints();

// Verifies uniform quantized tensor type scale and zp matches
LogicalResult verifyUniformQuantizedScaleAndZeroPointConstraints();
```

Reply:
done, thanks for the suggestion.